### PR TITLE
fix: replace halo with yaspin

### LINKED ISSUE #18: Unhandled error printout with cli client
## Describe the bug

I followed the docs for the cli version of the client (which works!), however it seems like an unhandled exception towards the end of that process keeps printing to the console

## To Reproduce

Followed steps from here: https://docs.fief.dev/integrate/python/cli/

```py
In [3]: client.authorize()
Out [3]: ...  # access_token stuff
Error in callback <function Halo.__init__.<locals>.clean_up at 0x744c898153a0> (for post_run_cell), with arguments args (<ExecutionResult object at 744c898249d0, ...
TypeError: Halo.__init__.<locals>.clean_up() takes 0 positional arguments but 1 was given
```

## Expected behavior

Doesn't appear to affect operation, but is annoying to see this error no matter what is happening lol

## Configuration

* Python version:3.11.9
* Fief client version: 0.18.6

* Fief Cloud or self-hosted: self-hosted (Railway)
* If self-hosted, Fief version: 0.28.6

## Additional context

Very sweet having a CLI client! (need it for our use case)

### PR DESCRIPTION
**Pull request by johnson2427**
## Description

Fixes #18 
This is a change to [fief-python](https://github.com/fief-dev/fief-python/tree/28b2dd61add4f69aef8db7204d64cba0f0627914)

Halo is used to produce the spinner in the cli and IPython. This causes an error, and since halo is no longer maintained, we have replaced halo with yaspin

### CODE REVIEW

**Review by fubuloubu — APPROVED** [review]

File: pyproject.toml
Comment (frankie567) [inline pyproject.toml]: The dependency should be in the `cli` group and replace `halo` (line 116).

**Review by frankie567 — CHANGES_REQUESTED** [review]
Thank you, looks good 🙏 The only change needed is to move the dependency in the right group (see comment above).

File: pyproject.toml
Comment (johnson2427) [reply pyproject.toml]: got it, should have searched for halo in the pyproject before pushing. Thank you!

Comment (johnson2427) [reply pyproject.toml]: before merging this, attempt the install. This isn't installing yaspin on my machine at the moment

**Comment by johnson2427** [comment]
Had an error in testing, I have all tests passing locally, hoping the last push gets us there

**Comment by frankie567** [comment]
Can you give me rights to edit the PR (**Allow edits from maintainer** on the right)? I've a fix for failing tests.

**Comment by fubuloubu** [comment]
> Can you give me rights to edit the PR (**Allow edits from maintainer** on the right)? I've a fix for failing tests.

We looked for a solid 20 minutes and can't figure out where github moved this option 😅 

Maybe it's some configuration we need to update, but github docs no help there either

**Comment by frankie567** [comment]
Normally, it's on the sidebar:

Capture d’écran 2024-06-03 à 16 38 25

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

**Comment by johnson2427** [comment]
This is what I have over there
image

**Comment by frankie567** [comment]
Ok! Editing PR from forks is always a nightmare 😅 I'll merge it as it and apply my changes on the main branch.

Thank you for the work on this 🙏

**Pull request merged**
